### PR TITLE
Fix slots, add declarative-shadow-dom if needed

### PR DESCRIPTION
**Pull request by Mihaiii**
For https://github.com/Starcounter/RebelsLounge/issues/152

### CODE REVIEW

File: src/SignIn/wwwroot/SignIn/viewmodels/AccessDeniedPage.html [hunk 1]
```diff
@@ -4,3 +4,7 @@
     </p>
     <a slot="SignIn/sign-in-link" href="/signin/signinuser?%2Fsignin%2Fprofile" class="btn btn-link">Sign in</a>
```
Comment (Mackiovello) [inline src/SignIn/wwwroot/SignIn/viewmodels/AccessDeniedPage.html]: I don't think the `href` value should be encoded. That's a topic for another PR

File: src/SignIn/wwwroot/SignIn/viewmodels/SignInPage.html [hunk 2]
```diff
@@ -179,4 +179,7 @@
             };
         })();
     </script>
+    <template is="declarative-shadow-dom">
+        <slot name="SignIn/current-user"></slot>
+    </template>
```
Comment (Mackiovello) [inline src/SignIn/wwwroot/SignIn/viewmodels/SignInPage.html]: This view is not blendable at all. Again, that's not for this PR

File: src/SignIn/wwwroot/SignIn/viewmodels/AccessDeniedPage.html [hunk 3]
```diff
@@ -3,4 +3,8 @@
         You do not have permissions to access this page!
     </p>
     <a slot="SignIn/sign-in-link" href="/signin/signinuser?%2Fsignin%2Fprofile" class="btn btn-link">Sign in</a>
+    <template is="declarative-shadow-dom">
+        <slot name="SignIn/access-denied-message"></slot>
```
Comment (PatrykSzwer) [inline src/SignIn/wwwroot/SignIn/viewmodels/AccessDeniedPage.html]: if I remember correctly We agreed to use only small case in naming slots. I discussed this with @warpech  during adding slots in kitchensink many weeks ago. Still I guess It's not topic of this PR.

File: src/SignIn/wwwroot/SignIn/viewmodels/AccessDeniedPage.html (hunk 1 above)
Comment (warpech) [inline src/SignIn/wwwroot/SignIn/viewmodels/AccessDeniedPage.html]: It is correct. The query part (after `?`) should be encoded. See: https://stackoverflow.com/questions/25979802/should-i-url-encode-a-query-string-parameter-thats-an-url

**Pull request merged**